library_generation/new_client/new-client.py

```diff
@@ -0,0 +1,468 @@
+         "needed or not."
+)
+@click.option(
+    "--proto-path",
+    required=True,
+    type=str,
+    default=None,
+    help="Path to proto file from the root of the googleapis repository to the"
+         "directory that contains the proto files (without the version)."
+         "For example, to generate the library for 'google/maps/routing/v2', "
+         "then you specify this value as 'google/maps/routing'"
+)
+@click.option(
+    "--cloud-api",
+    type=bool,
+    default=True,
+    show_default=True,
+    help="If true, the artifact ID of the library is 'google-cloud-'; "
+         "otherwise 'google-'"
+)
+@click.option(
+    "--group-id",
+    type=str,
+    default="com.google.cloud",
+    show_default=True,
+    help="The group ID of the artifact when distribution name is not set"
+)
+@click.option(
+    "--library-type",
+    type=str,
+    default="GAPIC_AUTO",
+    show_default=True,
+    help="A label that appear in repo-metadata.json to tell how the library is "
+         "maintained or generated"
+)
+@click.option(
+    "--googleapis-url",
+    type=str,
+    default="https://github.com/googleapis/googleapis.git",
+    show_default=True,
+    help="The URL of the repository that has proto service definition"
+)
+@click.option(
+    "--rest-docs",
+    type=str,
+    help="If it exists, link to the REST Documentation for a service"
+)
+@click.option(
+    "--rpc-docs",
+    type=str,
+    help="If it exists, link to the RPC Documentation for a service"
+)
+@click.option(
+    "--versions-file",
+    type=str,
+    default="output/versions.txt",
+    help="A text file contains versions of modules"
+)
+@click.option(
+    "--split-repo",
+    type=bool,
+    default=False,
+    help="Whether generating a library into a split repository"
+)
+def generate(
+    api_shortname,
+    name_pretty,
+    product_docs,
+    api_description,
+    generator_version,
+    release_level,
+    distribution_name,
+    api_id,
+    requires_billing,
+    transport,
+    language,
+    proto_path,
+    cloud_api,
+    group_id,
+    library_type,
+    googleapis_url,
+    rest_docs,
+    rpc_docs,
+    versions_file,
+    split_repo,
+):
+    cloud_prefix = "cloud-" if cloud_api else ""
+
+    destination_path = f"{language}-{api_shortname}"
+
+    if distribution_name is None:
+        distribution_name = f"{group_id}:google-{cloud_prefix}{api_shortname}"
+
+    distribution_name_short = re.split(r"[:\/]", distribution_name)[-1]
+
+    if api_id is None:
+        api_id = f"{api_shortname}.googleapis.com"
+
+    if not product_docs.startswith("https"):
+        sys.exit("product_docs must starts with 'https://'")
+
+    client_documentation = (
+        f"https://cloud.google.com/{language}/docs/reference/{distribution_name_short}/latest/overview"
+    )
+
+    if api_shortname == "":
+        sys.exit("api_shortname is empty")
+
+    repo = "googleapis/google-cloud-java"
```

Comment:
This is the right value since we only use the script to generate library inside the monorepo.